### PR TITLE
[WIP] Add deb_to_rb.sh

### PR DESCRIPTION
**Pull request by supechicken**
A script to convert `.deb` files into `.rb`package script

Tested with `microsoft-edge-dev`, `opera`

Usage :
```shell
bash deb_to_rb.sh <deb download URL> /usr/local/lib/crew/packages/
```

### CODE REVIEW

**Comment by uberhacker** [comment]
Wow @supechicken, this is ambitious!

File: tools/deb_to_rb.sh [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+if [ -d usr/local/bin/ ]; then echo '      FileUtils.mv "usr/local/bin/", CREW_DEST_PREFIX' >> ${out}; fi
+
+if [ -d usr/sbin/ ]; then echo '      FileUtils.mv "usr/sbin/", CREW_DEST_PREFIX' >> ${out}; fi
+if [ -d sbin/ ]; then echo '      FileUtils.mv "sbin/", CREW_DEST_PREFIX' >> ${out}; fi
+if [ -d usr/local/sbin/ ]; then echo '      FileUtils.mv "usr/local/sbin/", CREW_DEST_PREFIX' >> ${out}; fi
+
+if [ -d usr/share/ ]; then echo '      FileUtils.mv "usr/share/", CREW_DEST_PREFIX' >> ${out}; fi
+if [ -d usr/local/share/ ]; then echo '      FileUtils.mv "usr/local/share/", CREW_DEST_PREFIX' >> ${out}; fi
+
+if [ -d usr/opt/ ]; then echo '      FileUtils.mv "usr/opt/", CREW_DEST_PREFIX}/opt/' >> ${out}; fi
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:127]: Extra `}` after `CREW_DEST_PREFIX`?

File: tools/deb_to_rb.sh [hunk 2]
```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:1]: Would it make sense to add switch options to add the name, description, homepage, etc.?

File: tools/deb_to_rb.sh [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+ar x deb
+
+
+tar xvf control.tar.*
+  
+name=$(grep "Package: " ./control | sed -e 's/Package: //')
+pkgname=$(echo ${name} | sed -e 's/-/_/g')
+ver=$(grep "Version: " ./control | sed -e 's/Version: //' | sed -e 's/-/_/g')
+info=$(grep "Description: " ./control | sed -e 's/Description: //' | sed -e 's/-/_/g')
+sha256sum=$(sha256sum ./deb | cut -d ' ' -f1)
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:35]: Would it make sense to add switch options to add the name, description, homepage, etc.?

File: tools/deb_to_rb.sh (hunk 1 above)
Comment (supechicken) [reply tools/deb_to_rb.sh]: Removed

File: tools/deb_to_rb.sh [hunk 4]
```diff
@@ -0,0 +1,144 @@
+#!/bin/bash
+
+set -e
+
+CREW_PREFIX=/usr/local
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:5]: Use environment variable, if set?

File: tools/deb_to_rb.sh [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+CREW_LIB_PREFIX=${CREW_PREFIX}/lib
+pkg_prefix=${CREW_LIB_PREFIX}/crew/packages
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:7]: Can we be consistent and capitalize all variables?

File: tools/deb_to_rb.sh [hunk 6]
```diff
@@ -0,0 +1,144 @@
...
+GREEN='\e[1;32m';
+RED='\e[1;31m';
+BLUE='\e[1;34m';
+YELLOW='\e[1;33m';
+RESET='\e[0m'
+
+rm -rf ${out}
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:14]: Where is `${out}` set earlier in the script?

File: tools/deb_to_rb.sh (hunk 4 above)
Comment (supechicken666) [reply tools/deb_to_rb.sh]: Done

File: tools/deb_to_rb.sh (hunk 5 above)
Comment (supechicken666) [reply tools/deb_to_rb.sh]: Done

File: tools/deb_to_rb.sh (hunk 6 above)
Comment (supechicken666) [reply tools/deb_to_rb.sh]: Removed

File: tools/deb_to_rb.sh [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+BLUE='\e[1;34m';
+YELLOW='\e[1;33m';
+RESET='\e[0m'
+
+rm -rf ${CREW_PREFIX}/tmp/deb && cd ${CREW_PREFIX}/tmp/
+
+mkdir -p deb && cd ./deb
+
+echo -e "${YELLOW}Downloading deb...${RESET}"
+curl '-#' ${1} -o deb
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:19]: Shouldn't we check if `${1}` exists first and output a usage message if not?

File: tools/deb_to_rb.sh [hunk 8]
```diff
@@ -0,0 +1,140 @@
...
+
+ar x deb
+
+
+tar xf control.tar.*
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:24]: Is every extracted deb file named `control.tar.*`?

File: tools/deb_to_rb.sh [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+INFO=$(grep "Description: " ./control | sed -e 's/Description: //' | sed -e 's/-/_/g')
+SHA256SUM=$(sha256sum ./deb | cut -d ' ' -f1)
+
+OUT=${2}/${PKGNAME}.rb
+  
+cat <<EOF> ${OUT}
+require 'package'
+class ${PKGNAME^} < Package
+  description '${INFO}'
+  homepage ''
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:38]: Every package should include a homepage.  Maybe prompt for the url if it hasn't been passed as an argument?

File: tools/deb_to_rb.sh [hunk 10]
```diff
@@ -0,0 +1,140 @@
...
+PKGNAME=$(echo ${name} | sed -e 's/-/_/g')
+VER=$(grep "Version: " ./control | sed -e 's/Version: //' | sed -e 's/-/_/g')
+INFO=$(grep "Description: " ./control | sed -e 's/Description: //' | sed -e 's/-/_/g')
+SHA256SUM=$(sha256sum ./deb | cut -d ' ' -f1)
+
+OUT=${2}/${PKGNAME}.rb
+  
+cat <<EOF> ${OUT}
+require 'package'
+class ${PKGNAME^} < Package
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:36]: Why is the `^` needed?

Comment (uberhacker) [reply tools/deb_to_rb.sh:36]: OK, I understand now.  It's used to uppercase the first character.

File: tools/deb_to_rb.sh [hunk 11]
```diff
@@ -0,0 +1,140 @@
...
+         sed -e 's/libx11/sommelier/g' |\
+         sed -e 's/libgcc1/gcc10/g' |\
+         sed -e 's/libpangocairo/pango, cairo/g' |\
+         sed -e 's/libgtk_3/gtk3/g' |\
+         sed -e 's/libglib/glib/g' |\
+         sed -e 's/libgtk_2/gtk2/g' |\
+         sed -e 's/libatspi...../at_spi2_core/g' |\
+         sed -e 's/libqt5core5a/qtbase/g' |\
+         sed -e 's/\.//g' |\
+         sed -e 's/\,//g'))
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:64]: I imagine this search and replace logic is not complete.

File: tools/deb_to_rb.sh [hunk 12]
```diff
@@ -0,0 +1,140 @@
...
+  end
+    
+  def self.install
+    Dir.chdir '..' do
+      FileUtils.mkdir_p CREW_DEST_PREFIX
+      ENV['CREW_NOT_STRIP'] = '1'
+EOF"
+echo -e "${YELLOW}Unpacking 'data.tar' using 'tar', this may take a while...${RESET}"
+tar xvf data.tar.* > /dev/null
+if [ -d usr/bin ]; then echo '      FileUtils.mv "usr/bin/", CREW_DEST_PREFIX' >> ${OUT}; fi
```
Comment (uberhacker) [inline tools/deb_to_rb.sh:118]: FileUtils is for `.rb` files, not `.sh`.

File: tools/deb_to_rb.sh (hunk 8 above)
Comment (supechicken) [reply tools/deb_to_rb.sh:24]: Yeah, that file may be 'control.tar.xz' or 'control.tar.gz'

File: tools/deb_to_rb.sh (hunk 11 above)
Comment (supechicken) [reply tools/deb_to_rb.sh:73]: We need more filters to fit the dependency name to `crew` one

**Pull request closed without merging**